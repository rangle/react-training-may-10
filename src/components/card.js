import React from 'react';
import PropTypes from 'proptypes';
import { Media } from './media';
import { Heading } from './heading';
import { SubHeading } from './sub-heading';

export const Card = ({ name, id, email }) => (
  <article className="w5 mr3 bg-light-green br3 pa3 pa4-ns mv3 ba b--black-50">
    <div className="tc">
      <Media
        src={`https://robohash.org/${id}`}
        title='photo of robot'
      />
      <Heading>{ name }</Heading>
      <SubHeading>{ email }</SubHeading>
    </div>
  </article>
);

Card.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  email: PropTypes.string.isRequired,
};
