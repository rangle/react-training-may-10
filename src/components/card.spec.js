import React from 'react';
import renderer from 'react-test-renderer';
import { Card } from './card';

test('Card renders correctly', () => {
  const tree = renderer.create(
    <Card name="Ervin Howell" id={0} email="Shanna@melissa.tv" />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
