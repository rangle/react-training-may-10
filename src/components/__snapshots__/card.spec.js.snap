exports[`test Card renders correctly 1`] = `
<article
  className="w5 mr3 bg-light-green br3 pa3 pa4-ns mv3 ba b--black-50">
  <div
    className="tc">
    <img
      alt=""
      className="br-100 h4 w4 dib ba b--black-50 pa2"
      src="https://robohash.org/0"
      title="photo of robot" />
    <h1
      className="f3 mb2">
      Ervin Howell
    </h1>
    <h2
      className="f5 fw4 gray mt0">
      Shanna@melissa.tv
    </h2>
  </div>
</article>
`;
